packages/dev/src/main.ts

```diff
@@ -123,6 +135,16 @@ export class NetlifyDev {
         })
       : null
 
+    // Headers
+    const headers = this.#features.headers
+      ? new HeadersHandler({
+          configPath: this.#config?.configPath,
+          configHeaders: this.#config?.config.headers,
+          projectDir: this.#projectRoot,
+          publishDir: this.#config?.config.build.publish ?? undefined,
+        })
+      : null
```

Comment:
nit: Since you're checking multiple times whether `headers` exist and conditionally calling it, I would consider defining a no-op alternate here so that you can always call `headers.handle` without checking each time.

```suggestion
      : { handle: (_: Request, res: Response) => res }
```